plugins/utils/dressing.nix

```diff
@@ -0,0 +1,428 @@
+              of total."
+            '';
+
+          height = helpers.defaultNullOpts.mkNullable intOrRatio "null" ''
+            Can be an integer or a float between 0 and 1 (e.g. 0.4 for 40%).
+          '';
+
+          max_height =
+            helpers.defaultNullOpts.mkNullable
+            (with types; either intOrRatio (listOf intOrRatio)) "0.9"
+            ''
+              Max height of window.
+
+              Can be a list of mixed types, e.g. `[140 0.8]` means "less than 140 rows or 80%
+              of total."
+            '';
+
+          min_height =
+            helpers.defaultNullOpts.mkNullable
+            (with types; either intOrRatio (listOf intOrRatio)) "[10 0.2]"
+            ''
+              Min height of window.
+
+              Can be a list of mixed types, e.g. `[10 0.2]` means "less than 10 rows or 20%
+              of total."
+            '';
+
+          mappings =
+            helpers.defaultNullOpts.mkAttrsOf (with types; either str (enum [false]))
+            ''
+              {
+                "<Esc>" = "Close";
+                "<C-c>" = "Close";
+                "<CR>" = "Confirm";
+              }
+            ''
+            ''
+              Mappings in normal mode for the builtin selector.
+
+              To disable a default mapping in a specific mode, set it to `false`.
+            '';
+
+          override =
+            helpers.defaultNullOpts.mkLuaFn
+            ''
+              "function(conf) return conf end"
+            ''
+            ''
+              Lua function that takes config that is passed to nvim_open_win.
+              Used to customize the layout.
+            '';
+        };
+
+        format_item_override =
+          helpers.defaultNullOpts.mkAttrsOf helpers.nixvimTypes.strLuaFn "{}"
+          ''
+            Override the formatting/display for a specific "kind" when using vim.ui.select.
+            For example, code actions from vim.lsp.buf.code_action use a kind="codeaction".
+            You can override the format function when selecting for that kind, e.g.
+            format_item_override = {
+              codeaction = \'\'
+                function(action_tuple)
+                  local title = action_tuple[2].title:gsub("\r\n", "\\r\\n")
+                  local client = vim.lsp.get_client_by_id(action_tuple[1])
+                  return string.format("%s\t[%s]", title:gsub("\n", "\\n"), client.name)
+                end
+               \'\';
+            }
+          '';
+
+        get_config = helpers.defaultNullOpts.mkLuaFn "null" ''
+          This can be a function that accepts the opts parameter that is
+          passed in to 'vim.select' or 'vim.input'. It must return either
+          nil or config values to use in place of the global config values
+          for that module.
+
+          See :help dressing_get_config for more info.
+        '';
+      };
+    };
+
+    settingsExample = {
```

Comment:
This is way too long.
`settingsExample` is used for documentation generation (see [here](https://nix-community.github.io/nixvim/plugins/oil/settings/index.html#pluginsoilsettings) for an example).
It should be at most 10-15 lines.
If needed, you can add an `example` test case in the test file.

Reply:
done (just kept the most important functionality)